src/index.js

```diff
@@ -0,0 +1,517 @@
+  }
+}
+
+
+let applyMode = applyToAttr;
+let uniqueId = 0;
+
+
+function upgrade(node) {
+  const effectiveParent = node['scoped'] && document.body.contains(node) ? node.parentNode : null;
+
+  const state = styleNodes.get(node);
+  if (state) {
+    if (!effectiveParent) {
+      // disappearing, clear state and ask browser to reset CSS
+      styleNodes.delete(node);
+      resetCSS(node);
+    } else if (node.sheet) {
+      // otherwise, upgrade the sheet (succeeds if already done)
+      // nb. node.sheet is null if being removed
+      upgradeSheet(node.sheet, state.prefix);
+    }
+
+    if (state.parent !== effectiveParent) {
+      state.parent && applyMode(state.parent, state.attrName, false);
+      effectiveParent && applyMode(effectiveParent, state.attrName, true);
+      state.parent = effectiveParent;
+    }
+
+    return false;  // already upgraded
+  }
+
+  if (!effectiveParent) {
+    return;  // not scoped CSS, never seen before, ignore
+  }
+
+  // TODO: use hash for deduping
+  // const hash = hashCode(node.textContent);
+
+  // newly found style node, setup attr
+  const attrName = `${scopedCSSOptions['prefix']}${++uniqueId}`
+  const prefix = applyMode === applyToAttr ? `[${attrName}]` : `.${attrName}`;
+  styleNodes.set(node, {attrName, prefix, parent: node.parentNode});
+
+  upgradeSheet(node.sheet, prefix);
+  applyMode(effectiveParent, attrName, true);
+}
+
+// this mess basically calls resolve() with any <style> nodes that changed/removed/added
+// TODO: is it faster to just call getElementsByTagName('style') and compare to previous
+const mo = new MutationObserver((records) => {
+  const changes = new Set();
+  const iterate = (nodes) => {
+    let i = nodes ? nodes.length : 0;
+    while (i) {
+      const node = nodes[--i];
+      if (!(node instanceof HTMLElement)) {
+        continue;  // text node
+      }
+      if (node instanceof HTMLStyleElement) {
+        changes.add(node);  // directly a <style>
+        continue;
+      }
+      // look for changed children
+      const cand = node.getElementsByTagName('style');
+      let j = cand.length;
+      while (j) {
+        changes.add(cand[--j]);
+      }
+    }
+  };
+
+  records.forEach((record) => {
+    if (record.target instanceof HTMLStyleElement) {
+      changes.add(record.target);
+    } else {
+      iterate(record.addedNodes);
+      iterate(record.removedNodes);
+    }
+  });
+
+  changes.forEach(upgrade);
+});
+
+export function setup() {
+  // clone any options from global
+  const cand = window['scopedCSS'];
+  if (typeof cand === 'object') {
+    for (let k in scopedCSSOptions) {
+      if (k in cand) {
+        scopedCSSOptions[k] = cand[k];
+      }
+    }
+
+    if (scopedCSSOptions['applyToClass']) {
+      applyMode = applyToClass;
+    }
+  }
+
+  // nb. watch for attributeFilter: ['scoped'] to detect a CSS rule changing at runtime
+  const options = {childList: true, subtree: true, attributes: true, attributeFilter: ['scoped']};
```

Comment:
not important for now, but I wonder if this actually detects content changes in style tags, or just the `scoped` attribute toggling...

Reply:
I've played it around by randomly changing a style tag's body and it works for me, will add a few tests to further confirm it.